syntax = "proto3";

package psyneulinkviewer;


service ServeGraph {
    rpc LoadModel (ModelPath) returns (GraphJson) {}
    rpc UpdateModel (GraphJson) returns (Response) {}
    rpc RunModel (InputJson) returns (Response) {}
    rpc PNLApi (PNLJson) returns (PNLJson) {}
}


enum ResponseMessage {
    UP_AND_RUNNING = 0;
    MESSAGE_OK = 1;
    MESSAGE_ERROR = 2;
    CLOSED_CONNECTION = 3;
}


message NullArgument {}

message Response {
    ResponseMessage response = 1;
    string message = 2;
}

message ModelPath {
    string path = 1;
}

message GraphJson {
    string modelJson = 1;
}

message PNLJson {
    string genericJson = 1;
}

message InputJson {
    string executableNode = 1;
    string inputData = 2;
}
